File: Problems/InvertBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {    
    const auxStack = [root];
    
    while(auxStack.length){        
        const node = auxStack.pop();
        
        if(node != null){            
            auxStack.push(node.left);
            auxStack.push(node.right);
            
            const temp = node.left;
            node.left = node.right;
            node.right = temp;
        }
    }
    return root;
};

File: DataStructure/InvertBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None
        
        invertedTree = TreeNode(root.val)        
        
        invertedTree.left = self.invertTree(root.right)
        invertedTree.right = self.invertTree(root.left)
        
        return invertedTree
